google/cloud/pubsub/internal/tracing_message_callback.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/internal/message_callback.h"
+#include "google/cloud/pubsub/options.h"
+#include "google/cloud/pubsub/subscription.h"
+#include "google/cloud/pubsub/version.h"
+#include "google/cloud/internal/opentelemetry.h"
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+#include <google/pubsub/v1/pubsub.pb.h>
```

Comment:
Unused. Also, for consistency, let's only put `<opentelemetry/...>` headers in the `#ifdef` block.